src/Events/Admin/Onboarding/Controller.php

```diff
@@ -0,0 +1,167 @@
+<?php
+/**
+ * Controller for interfacing with TEC\Common\Onboarding.
+ *
+ * @since 7.0.0
+ */
+
+namespace TEC\Events\Admin\Onboarding;
+
+use TEC\Common\Contracts\Provider\Controller as Controller_Contract;
+use TEC\Events\Admin\Onboarding\API;
+use TEC\Events\Admin\Onboarding\Steps\Optin;
+use TEC\Events\Admin\Onboarding\Steps\Settings;
+use TEC\Events\Admin\Onboarding\Steps\Organizer;
+use TEC\Events\Admin\Onboarding\Steps\Venue;
+use TEC\Events\Admin\Onboarding\Steps\Tickets;
+use TEC\Events\Admin\Onboarding\Data;
+use TEC\Events\Admin\Onboarding\Landing_Page;
+
+/**
+ * Class Controller
+ *
+ * @since 7.0.0
+ * @package TEC\Events\Admin\Onboarding
+ */
+class Controller extends Controller_Contract {
+
+	/**
+	 * Register the provider.
+	 *
+	 * @since 7.0.0
+	 */
+	public function do_register(): void {
+		$this->add_filters();
+		$this->add_actions();
+
+		// Register the service provider itself on the container.
+		$this->container->singleton( static::class, $this );
```

Comment:
I dont like a controller registering itself. He should be registered where he is being initialized instead. Why don't we go one controller up and register this controller there ?

Keep in mind that `->register` actually creates a `->singleton`